strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/services/support/ArtifactStoredEventListener.java

```diff
@@ -36,34 +37,32 @@ public ArtifactStoredEventListener()
     }
 
     @Override
-    protected ArtifactEntry handleEvent(RepositoryPath repositoryPath)
+    protected Artifact handleEvent(RepositoryPath repositoryPath)
             throws IOException
     {
-        ArtifactEntry artifactEntry = repositoryPath.getArtifactEntry();
+        Artifact artifactEntry = repositoryPath.getArtifactEntry();
 
         if (artifactEntry == null)
         {
             logger.warn("No [{}] for [{}].",
-                        ArtifactEntry.class.getSimpleName(),
+                        Artifact.class.getSimpleName(),
                         repositoryPath);
 
             return null;
         }
 
-        final Repository repository = repositoryPath.getRepository();
-        final LayoutProvider layoutProvider = layoutProviderRegistry.getProvider(repository.getLayout());
-        final Set<String> archiveFilenames = layoutProvider.listArchiveFilenames(repositoryPath);
+        Repository repository = repositoryPath.getRepository();
+        LayoutProvider layoutProvider = layoutProviderRegistry.getProvider(repository.getLayout());
+        Set<String> archiveFilenames = layoutProvider.listArchiveFilenames(repositoryPath);
         if (archiveFilenames.isEmpty())
         {
             return null;
+        } else if (archiveFilenames.size() > 5) {
+            //TODO: issues/1752 
+            archiveFilenames = archiveFilenames.stream().limit(100).collect(Collectors.toSet());
```

Comment:
Cosmetic: Braces on new lines please. :)
```suggestion
        }
        else if (archiveFilenames.size() > 5)
        {
            //TODO: issues/1752 
            archiveFilenames = archiveFilenames.stream().limit(100).collect(Collectors.toSet());
```